### PR TITLE
Updates to AIDA TLU, Caribou & Timepix3 Producers

### PR DESCRIPTION
**Pull request by simonspa**
This PR contains fixes to the AIDA TLU, Caribou and Timepix3 producers as well as the respective data collectors:

* AIDA TLU: Support for new AIDA TLU firmware 0x25
* Caribou: allow to detect T0 signals and abort
* Caribou: allow to stack events into sub-events to not saturate link and data collector with packets
* Timepix3: allow to detect T0 signals and abort

One change to central classes: StandardEventConverter now provides a few exceptions to throw which allows a more detailed communication between calling code and the converter. Now it is for example possible to abort further processing because invalid data has been detected.

* Please do not squash commits when merging
* Please tag a new EUDAQ2 version v2.4.4 if possible so we can rely on available functionality 

(cc) @jekroege @lhuth

### CODE REVIEW

File: user/tlu/misc/aida_tlu/aida_tlu_address-fw_version_25.xml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  <!--<node id="TriggerPatternW" address="0x3" permission="w" description="" />-->
+  <!--<node id="TriggerPatternR" address="0x13" permission="r" description="" />-->
+  <node id="TriggerVetoW" address="0x4" permission="w" description="" />
+  <node id="TriggerVetoR" address="0x14" permission="r" description="" /><!--Wait, this does nothing at the moment...-->
+  <node id="ExternalTriggerVetoR" address="0x15" permission="r" description="" />
+  <node id="PulseStretchW" address="0x6" permission="w" description="" />
+  <node id="PulseStretchR" address="0x16" permission="r" description="" />
+  <node id="PulseDelayW" address="0x7" permission="w" description="" />
+  <node id="PulseDelayR" address="0x17" permission="r" description="" />
+  <node id="TriggerHoldOffW" address="0x8" permission="W" description="" /><!--Wait, this does nothing at the moment...-->
```
Comment (eyiliu) [inline user/tlu/misc/aida_tlu/aida_tlu_address-fw_version_25.xml:78]: The capitcal 'W' in permission is not legal in recent cactus/ipbus.

File: user/tlu/hardware/src/AidaTluController.cc [hunk 2]
```diff
@@ -607,7 +607,7 @@ namespace tlu {
   /// Writes to IPBus register to instruct the TLU to start/stop a run.
   //  state= 1 when running, 0 when stopped.
   //  On start, the internal registers are reset and the TLU issues a T0 pulse to DUTs.
-    SetWRegister("Shutter.RunActive", state);
+    SetWRegister("Shutter.RunActiveRW", state);
```
Comment (eyiliu) [inline user/tlu/hardware/src/AidaTluController.cc:610]: The current  registers xml version 10 using the name _Shutter.RunActive_ , not sure if we should also correct version 10 xml.

File: user/tlu/module/src/TluRawEvent2StdEventConverter.cc [hunk 3]
```diff
@@ -30,20 +29,36 @@ bool TluRawEvent2StdEventConverter::Converting(eudaq::EventSPC d1, eudaq::Standa
...
+               + (finets3 * ((triggersFired >> 3) & 0x1))
+               + (finets4 * ((triggersFired >> 4) & 0x1))
+               + (finets5 * ((triggersFired >> 5) & 0x1)))
+               / __builtin_popcount(triggersFired & 0x3F); // count "ones" in binary
+
+  auto ts = (d1->GetTimestampBegin() * 1000) & 0xFFFFFFFFFFFFFF00 + static_cast<uint64_t>(25. / 32. * finets);
+
+  d2->SetTimeBegin(ts);
   d2->SetTimeEnd(d1->GetTimestampEnd() * 1000);
+  d2->SetTimestamp(ts, d1->GetTimestampEnd(), d1->IsFlagTimestamp());
```
Comment (eyiliu) [inline user/tlu/module/src/TluRawEvent2StdEventConverter.cc:57]: line 55 56 could be not necessary  it  the line 57 always override them.

File: user/tlu/hardware/src/AidaTluController.cc (hunk 2 above)
Comment (simonspa) [reply user/tlu/hardware/src/AidaTluController.cc:610]: This seems to be a change from the latest FW only, at least according to the register information and errors printed when initializing.

File: user/tlu/module/src/TluRawEvent2StdEventConverter.cc (hunk 3 above)
Comment (simonspa) [reply user/tlu/module/src/TluRawEvent2StdEventConverter.cc:57]: I beg to differ:

* `SetTimestamp()` is a member of `eudaq::Event` and sets a timestamp in clock ticks.
* `SetTimeBegin()` and `SetTimeEnd()` are members of `eudaq::StandardEvent` and set absolute times in picoseconds.

The two variables are complementary.

File: user/tlu/misc/aida_tlu/aida_tlu_address-fw_version_25.xml (hunk 1 above)
Comment (simonspa) [reply user/tlu/misc/aida_tlu/aida_tlu_address-fw_version_25.xml:78]: Fixed, also retrospectively for every configuration ever used and committed to the repo.

File: user/tlu/hardware/src/AidaTluController.cc (hunk 2 above)
Comment (lhuth) [reply user/tlu/hardware/src/AidaTluController.cc:610]: @simonspa is right - took us a few minutes to figure it out.

**Pull request merged**